scipy/stats/_qmc.py

```diff
@@ -1,33 +1,47 @@
 """Quasi-Monte Carlo engines and helpers."""
-import os
+from __future__ import annotations
+
 import copy
-import numbers
-from abc import ABC, abstractmethod
 import math
+import numbers
+import os
 import warnings
+from abc import ABC, abstractmethod
+from typing import (
+    Optional,
+    Union,
+)
 
 import numpy as np
-
-import scipy.stats as stats
-from scipy.stats._sobol import (
-    initialize_v, _cscramble, _fill_p_cumulative, _draw, _fast_forward,
-    _categorize, initialize_direction_numbers, _MAXDIM, _MAXBIT
-)
 from scipy.stats._qmc_cy import (
     _cy_wrapper_centered_discrepancy,
     _cy_wrapper_wrap_around_discrepancy,
     _cy_wrapper_mixture_discrepancy,
     _cy_wrapper_l2_star_discrepancy,
     _cy_wrapper_update_discrepancy
 )
+from scipy.stats._sobol import (
+    initialize_v, _cscramble, _fill_p_cumulative, _draw, _fast_forward,
+    _categorize, initialize_direction_numbers, _MAXDIM, _MAXBIT
+)
+
+import scipy.stats as stats
+
 
 __all__ = ['scale', 'discrepancy', 'update_discrepancy',
            'QMCEngine', 'Sobol', 'Halton', 'LatinHypercube',
            'MultinomialQMC', 'MultivariateNormalQMC']
 
+_IntegerType = Union[int, np.integer]
```

Comment:
Call these `IntNumber` and `DecimalNumber` or something more non-code? It gets really crowded towards the end of this file.

Reply:
Changes are in 10ba2affd279d2bbce932cdeadfbbe6340876eea. Thank you @ilayn